pitest-entry/src/main/java/org/pitest/mutationtest/build/intercept/javafeatures/RecordFilter.java

```diff
@@ -0,0 +1,61 @@
+package org.pitest.mutationtest.build.intercept.javafeatures;
+
+import org.objectweb.asm.tree.RecordComponentNode;
+import org.pitest.bytecode.analysis.ClassTree;
+import org.pitest.mutationtest.build.InterceptorType;
+import org.pitest.mutationtest.build.MutationInterceptor;
+import org.pitest.mutationtest.engine.Mutater;
+import org.pitest.mutationtest.engine.MutationDetails;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class RecordFilter implements MutationInterceptor {
+
+    private boolean isRecord;
+    private ClassTree currentClass;
+
+    @Override
+    public InterceptorType type() {
+        return InterceptorType.FILTER;
+    }
+
+    @Override
+    public void begin(ClassTree clazz) {
+        isRecord = "java/lang/Record".equals(clazz.rawNode().superName);
```

Comment:
Just FYI, the `Class<T>` type also provides a simple flag to indicate if a type is a record without checking the superclass name: https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Class.html#isRecord() not sure if that's easier to use here or if it can be exposed by the `ClassTree` type.

Reply:
We need to compile and run under java 8, is isRecord is not available.